docs/source/federated-types/composition.mdx

```diff
@@ -248,3 +248,255 @@ type User @key(fields: "id") {
+}
+
+union Media = Book | Podcast
+
+interface BookDetails {
+  title: String!
+  numPages: Int
+}
+```
+
+<></>
+
+</CodeColumns>
+
+When these subgraph schemas are composed, the composition process merges the three corresponding types using the union strategy. This results in the following type definitions in the supergraph schema:
+
+```graphql title="Supergraph schema"
+type User {
+  id: ID!
+  age: Int!
+  name: String!
+  email: String!
+}
+
+union Media = Book | Movie | Podcast
+
+interface BookDetails {
+  title: String!
+  author: String!
+  numPages: Int
+}
+```
+
+Because composition uses the union strategy for these types, subgraphs can contribute distinct parts and guarantee that those parts will appear in the composed supergraph schema.
+
+> Note that if different subgraphs contribute different fields to an interface type, any object types that _implement_ that interface must define _all_ contributed fields from _all_ subgraphs. Otherwise, composition fails.
+
+### Input types and field arguments
+
+Composition _always_ uses the intersection strategy to merge input types and field arguments. This makes sure that the gateway never passes an argument to a subgraph that doesn't define that argument.
+
+Consider the following subgraph schemas:
+
+<CodeColumns>
+
+```graphql title="Subgraph A"
+input UserInput {
+  name: String!
+  age: Int
+}
+
+type Library @shareable {
+  book(title: String, author: String): Book
+}
+```
+
+```graphql title="Subgraph B"
+input UserInput {
+  name: String!
+  email: String
+}
+
+type Library @shareable {
+  book(title: String, section: String): Book
+}
+
+```
+
+</CodeColumns>
+
+These subgraphs define different fields for the `UserInput` input type, and they define different arguments for the `Library.book` field. After composition uses the intersection strategy, the supergraph schema definitions look like this:
+
+```graphql title="Supergraph schema"
+input UserInput {
+  name: String!
+}
+
+type Library {
+  # field arguments are composed via the intersection merging strategy
+  findBook(title: String): Book
+}
+```
+
+As you can see, the supergraph schema includes only the input fields and arguments that _both_ subgraphs define.
+
+> ⚠️ **Important:** If the intersection strategy would omit an input field or argument that is _non-nullable_, **composition fails**. This is because at least one subgraph _requires_ that field or argument, and the gateway can't provide it if it's omitted from the supergraph schema.
+>
+> When defining input types and field arguments in multiple subgraphs, make sure that every non-nullable field and argument is consistent in _every subgraph_. For examples, see [Arguments](sharing-types#arguments).
+
+
+### Enums
+
+Composition might use the union strategy _or_ the intersection strategy when merging enum type definitions, _depending on how the enum is used_. 
+
+```mermaid
+graph TB;
+    parent("Is the enum used as the type for a field argument or input type field?");
```

Comment:
```suggestion
    enum("Is the enum used as the type for at least<br/>one field argument or input type field?");
```

Reply:
We can throw a line break into a mermaid bubble so it doesn't end up quite so wide

Reply:
I didn't know you could do that! 😲 Thank you!